src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -1186,32 +1214,31 @@ protected FormValidation doTestConnection(@AncestorInPath ItemGroup context, URL
                 return FormValidation.ok();
             }
             try {
-                SSHUserPrivateKey sshCredential = getSshCredential(sshKeysCredentialsId, context);
-                String privateKey = "";
-                if (sshCredential != null) {
-                    privateKey = sshCredential.getPrivateKey();
-                } else {
-                    return FormValidation.error("Failed to find credential \"" + sshKeysCredentialsId + "\" in store.");
-                }
-
                 AWSCredentialsProvider credentialsProvider = createCredentialsProvider(useInstanceProfileForCredentials, credentialsId, roleArn, roleSessionName, region);
                 AmazonEC2 ec2 = AmazonEC2Factory.getInstance().connect(credentialsProvider, ec2endpoint);
                 ec2.describeInstances();
```

Comment:
Is this needed?

Reply:
nope, i do not think so (that was in the original code) - i will remove it and test

Reply:
Maybe it's there just to check credentials are ok. Anyways, I thought it was added in this PR, if it's not, then let's leave it there.

Reply:
seems to function fine without it, and i reviewed the api call docs and there seems to be no purpose for it here.